### ISSUE TITLE
Sphinx autosummary modules should return module names without path

### ISSUE THREAD

**Issue by 4flixt** [sphinx-doc/sphinx#9743]
**Is your feature request related to a problem? Please describe.**
We are using ``sphinx.ext.autosummary`` with ``_templates`` to create new files for each class, attribute, function.
This works perfectly for the above mentioned, where the template file contains (e.g. just for ``classes``):

```rst
{% if classes %}
.. automodule:: {{ fullname }}
.. currentmodule:: {{ fullname }}

.. rubric:: Classes
.. autosummary::
    :toctree:
    :nosignatures:
    {% for class in classes %}
    {{ class }}
    {% endfor %}
{% endif %}
```

Note that we are setting with ``.. currentmodule:: {{ fullname }}`` the correct path to the current file and then document each class with: ``{% for class in classes %}``. This works, because ``classes`` returns the class names without prefix (the path contained in ``fullname``).

For modules, on the other hand, the approach doesn't work:

```rst
{% if modules %}
.. automodule:: {{ fullname }}
.. currentmodule:: {{ fullname }}
.. autosummary::
   :toctree:
   :recursive:
   :template: module.rst
   {% for module in modules %}
   {{ module }}
   {% endfor %}
{% endif %}
```

The problem is here that ``modules`` returns a list of modules with the **full path**. 

**Describe the solution you'd like**
Have ``modules`` return only the module name without prefix (full path)

**Additional context**
We use Sphinx to document our toolbox [do-mpc](www.do-mpc.com)

**Comment by JPEWdev** [comment]
I think https://github.com/sphinx-doc/sphinx/pull/6792 should raise the priority of this, because now Sphinx will reject the full path modules with an error. We had to work around this by doing:

```rst
{% if modules %}
.. rubric:: Modules

.. autosummary::
   :toctree:
   :template: module-template.rst
   :recursive:
{% for item in modules %}
   {{ item | replace(fullname + ".", "") }}
{%- endfor %}
{% endif %}
```

**Comment by AA-Turner** [comment]
Please see #12609, which ought resolve this.

A

**Comment by AA-Turner** [comment]
I have released [Sphinx 7.4.6](https://pypi.org/project/Sphinx/7.4.6/) with a fix.

A
